checks.d/yarn.py

```diff
@@ -146,61 +150,60 @@ def check(self, instance):
         # Get properties from conf file
         rm_address = instance.get('resourcemanager_uri', DEFAULT_RM_URI)
 
-        # Get the cluster ID from Yarn
-        cluster_id = self._get_cluster_id(rm_address)
-
-        # Get metrics from the Resource Manager
-        self._yarn_cluster_metrics(cluster_id, rm_address)
-        self._yarn_app_metrics(rm_address)
-        self._yarn_node_metrics(cluster_id, rm_address)
+        # Get additional tags from the conf file
+        tags = instance.get('tags', [])
+        if tags is None:
+            tags = []
+        else:
+            tags = list(set(tags))
 
-    def _get_cluster_id(self, rm_address):
-        '''
-        Return the cluster ID, otherwise raise an exception
-        '''
-        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+        # Get the cluster name from the conf file
+        cluster_name = instance.get('cluster_name')
+        if cluster_name is None:
+            raise Exception('The cluster_name must be specified in the instance configuration')
```

Comment:
The YARN check has already been released so it'd better to be compatible with the previous configuration format. Instead of raising an exception here could you use `self.warning("The cluster_name must be specified in the instance configuration, defaulting to 'default_cluster'")` and set the `cluster_name` to `default_cluster` by default?

This would let the user know that a `cluster_name` needs to be set for the check to work best, without making the check suddenly stop working when the agent is upgraded.